enterprise/server/test/integration/task_router/task_router_test.go

```diff
@@ -106,6 +112,113 @@ func TestTaskRouter_RankNodes_DefaultNodeLimit_ReturnsOnlyLatestNodeMarkedComple
 	require.Equal(t, executorID2, ranked[0].GetExecutorID())
 
 	requireNotAlwaysRanked(1, executorID1, t, router, ctx, cmd, instanceName)
+	requireNonSequential(t, ranked[1:])
+}
+
+func TestTaskRouter_RankNodes_AffinityRouting(t *testing.T) {
```

Comment:
I know it's sort of obvious (after reviewing the code) but it'd be nice to add a test for the expected happy path which is:

1) action runs
2) action inputs change
3) action runs again

to ensure you're hitting the same node

(apologies if i'm just missing the test where this happens)

Reply:
Good call, expanded this test case to do that